src/sync/arc.rs

```diff
@@ -29,9 +29,8 @@ impl<T> Arc<T> {
     }
 
     /// Gets the number of strong (`Arc`) pointers to this value.
-    pub fn strong_count(_this: &Self) -> usize {
-        unimplemented!("no tests checking this? DELETED!")
-        // this.inner.ref_cnt.load(SeqCst)
+    pub fn strong_count(this: &Self) -> usize {
+        std::sync::Arc::strong_count(&this.inner)
```

Comment:
I think this should return the value of the `Arc`'s modeled state's `ref_cnt` field: https://github.com/tokio-rs/loom/blob/9cd2cd21f015d2582308c705fe0e25c475e83cbb/src/rt/arc.rs#L14

And, I think this should synchronize the `Arc`'s state in the model with a `SeqCst` load, similarly to the other ref count ops: https://github.com/tokio-rs/loom/blob/73e6c14da41978e7a20e1993c2554a16a3dcb597/src/rt/arc.rs#L80-L81

That way, the returned value is actually part of the model, rather than bypassing it. In practice, I don't know if this is terribly important, but I think it's more correct.

Reply:
In practice, this is going to be a bit more complicated.

`Arc` implements its own "dependent access" tracking in order to be more efficient about reducing the state space. You will see in `rt/arc.rs` there is `enum Action` and this matters for the implementations of `last_dependent_access` and `set_last_access`. To implement `strong_count` correctly, we would need to update the type to manage the dependent access correctly.

The way to think about it is that two concurrent increments are not dependent, i.e. the order in which they execute does not impact the behavior of the program, so loom doesn't bother permuting it.

I believe a `strong_count` call is dependent with both incs and decs, but not other calls to `strong_count()`.

Reply:
Okay, thank you for the information :) I do not really know how loom works internally (I just began using it) so I will try find my way around it based on what's been said here.